File: Frontend/src/interfaces/message.ts
export interface Message {
    id: number,
    sender: User,
    date: Date,
    content: string,
    platform: Platform
    likeAmount: number
}

export interface User {
    id: number,
    name: string,
    email: string,
    dateOfBirth: Date
}

export enum Platform {
    FACEBOOK, TWITTER, LINKEDIN, INSTAGRAM, ISMO
}

export const PlatformName = {
    0: "Facebook",
    1: "Twitter",
    2: "LinkedIn",
    3: "Instagram",
    4: "ISMO"
}

File: Backend/IsmoBackend/Models/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace IsmoBackend.Models
{
    public class Message
    {
        public int Id { get; set; }
        public User Sender { get; set; }
        public DateTime Date { get; set; }
        public string Content { get; set; }
        public Platform Platform { get; set; }
        [NotMapped]
        public int LikeAmount { get; set; }
    }

    public enum Platform
    {
        FACEBOOK, TWITTER, LINKEDIN, INSTAGRAM, ISMO
    }
}
